src/in/shick/diode/markdown/Markdown.java

```diff
@@ -79,7 +79,7 @@
     static final RunAutomaton autoLinkUrlAutomaton = new RunAutomaton(new RegExp("((https?|ftp):([^'\"> \t\r\n])+)", RegExp.NONE).toAutomaton());
 //    static final Pattern autoLinkEmail = Pattern.compile("<([-.\\w]+\\@[-a-z0-9]+(\\.[-a-z0-9]+)*\\.[a-z]+)>");
 
-    static final RunAutomaton subredditAutomaton = new RunAutomaton(new RegExp("/[rR]/[a-zA-Z0-9]+/?", RegExp.NONE).toAutomaton());
+    static final RunAutomaton subredditAutomaton = new RunAutomaton(new RegExp("[rR]/[a-zA-Z0-9]+/?", RegExp.NONE).toAutomaton());
```

Comment:
Digging deeper, looks like the original author used a heavily limited regex library that's so speed optimized it doesn't even support the start-of-string anchor.  I guess that level of performance optimization might have been worth it back when RiF was new and phones were super slow.

So basically it's literally looking for a caret character. . . *sigh*.

Long-term it would be better to switch to a full regex engine, but for now I think it's fair to add your original version.  water/fire false positives are goofy, but they don't really hurt anything.